### PR TITLE
Adds fix for positioning, touch device support, and lint errors

### PR DESCRIPTION
**Pull request by estrattonbailey**
Hey! Thanks for the lib 👍  It was almost ready out of the box, but I noticed that it didn't account for page scroll when calculating the top and bottom values so I started digging. This PR:

1. Fixes lint errors encountered at build time
2. Calculates top and bottom values using `window.pageYOffset` (IE9+)
3. Disables the share dialog on touch devices

The last point is a little opinionated, and since touch is a fickle feature to detect, it's not totally foolproof, but I think it's the best way to ensure that the dialog doesn't compete with the native share dialogs on Android, iOS, etc.

Let me know if you'd like anything adjusted! Happy to help :)

### CODE REVIEW

**Comment by MaxArt2501** [comment]
Hello, thank you for your PR! 🎉
Yes, indeed, the last point is a bit opinionated as there's in fact no reliable way to determine if a device is touch-screen only or not. I can only check this Monday, but I fear that for a common touch-screen laptop this package wouldn't work. (Also, since that kind of detection would always give the same result, maybe the best place to put it would be the `init `method.)

The point is that I *mean* this package to work on touch devices too, even if they have native sharing capabilities. That's because one could provide additional sharers, even site-specific ones, that a device might not offer (e.g.: "save to my account's notes"). And also because *any* device, even non-touch ones, could offer native sharing features - not to mention the upcoming [Web Share API](https://blog.hospodarets.com/web-share-api) that could come to desktop/laptop PCs as well.

Being unopinionated means that limiting or extending the functionality of the package is up to the developer. That's also why I designed a (hopefully) simple system to add/remove sharers, and kept the HTML skeleton and CSS footprint to the bare minimum. About the mentioned case, one could disable `share-this` just by doing something like this:

```js
if ((("ontouchstart" in window) || (window.navigator && window.navigator.msPointerEnabled && window.MSGesture) || (window.DocumentTouch && document instanceof window.DocumentTouch))) {
    return;
}
const selectionShare = shareThis({ ... });
selectionShare.init();
```

But now that you've brought the argument up, it's indeed a good idea to mention it in the docs.

For the rest, yes, not taking the scroll position into account is a bug known to me. (Also considering *horizontal* scrolling would be nice - you might never know!) You have to consider this library has been left on hiatus for a while while trying to increase the tests' code coverage instead of, you know, *shipping stuff* (sorry 😭).

So, yes, there are some bugs, some things I'd like to refine, upgrade jsdom from v9 to v10, ... before giving a public notice on the usual channels. So I'm actually quite surprised that *anyone* is using it and also is taking his time to fork it and make a PR.

I think I'll leave this open for a while if you'd like to change something, otherwise I think I'll close it. In the meanwhile, I pinky-promise to work on it again soon (I *should* have some spare time).

**Comment by estrattonbailey** [comment]
Cheers, mate! All great points, and handling the touch events outside the handlers is a much better idea. If I get a minute this weekend I'll remove that commit. Would you be interested in merging the other two commits? I'm totally not offended if you plan to implement this stuff in your own way and want to just close this, no worries.

My 2¢ on tests: as a user I don't care too much about 100% test coverage on a lib like this, I just want it to work when I plug it in 😛 

Thanks for your work! It's a neat little lib, I especially like the pluggable sharer pattern.

**Comment by MaxArt2501** [comment]
Actually I have some other notes to add, I'll let you know in the comments on the committed code 😉

And yes, you're totally right about the tests, they're something that can wait a little.

File: src/core.js [hunk 1]
```diff
@@ -55,7 +55,10 @@ export default (opts) => {
             removeListener("touchend", selectionCheck);
 
             killPopover();
-            _selection = _window = _document = null;
+
+            _selection = null;
+            _window = null;
+            _document = null;
```
Comment (MaxArt2501) [inline src/core.js]: I have no idea why ESLint hasn't warned me about this...

File: src/selection.js [hunk 2]
```diff
@@ -39,8 +39,8 @@ export function getEndLineRect(range, isForward) {
     }
 
     return {
-        top: Math.min(...endLineRects.map(rect => rect.top)),
-        bottom: Math.max(...endLineRects.map(rect => rect.bottom)),
+        top: window.pageYOffset + Math.min(...endLineRects.map(rect => rect.top)),
+        bottom: window.pageYOffset + Math.max(...endLineRects.map(rect => rect.bottom)),
```
Comment (MaxArt2501) [inline src/selection.js]: I see two minor problems here:
1. `window` is a reference to the main window object, which might not be what we want (e.g. in a `<iframe>`);
2. the name `getEndLineRect` recalls other methods like `getClientRects` or `getBoundingClientRect`, where the "rects" are `ClientRect` objects, with their `top`, `right`, `bottom` and `left` properties being distances from the *viewport*'s borders: that's why this function doesn't add scrolling.

I think a better place to add vertical scrolling would be `stylePopover` in popover.js, where the style is actually applied. It also uses `getPageScroll` from utils.js which should provide a cross-browser way to retrieve the page scrolling (not that I care about IE8- anyway).

File: test/selection.js [hunk 3]
```diff
@@ -1,6 +1,5 @@
 /* eslint-disable consistent-return, no-undef, no-unused-expressions */
 import chai, { expect } from "chai";
-import { spy } from "sinon";
 import sinonChai from "sinon-chai";
 import { env } from "jsdom";
```
Comment (MaxArt2501) [inline test/selection.js]: Actually sinon isn't used in the whole test file, so we remove sinon-chai too and limit the imported stuff to `import { expect } from "chai";`.

**Comment by estrattonbailey** [comment]
I appreciate the feedback! I believe I have this sorted, let me know what you think 👍

**Comment by MaxArt2501** [comment]
Excellent, I will release v1.0.5 hopefully within this weekend.

**Pull request merged**